packages/lodestar/src/network/gossip/topic.ts

```diff
@@ -2,92 +2,121 @@
  * @module network/gossip
  */
 
-import {ContainerType, toHexString} from "@chainsafe/ssz";
+import {ContainerType} from "@chainsafe/ssz";
 import {IBeaconConfig} from "@chainsafe/lodestar-config";
-import {IForkDigestContext} from "../../util/forkDigestContext";
+import {IForkDigestContext, toHexStringNoPrefix} from "../../util/forkDigestContext";
+import {DEFAULT_ENCODING} from "./constants";
 import {
   GossipEncoding,
   GossipDeserializer,
   GossipObject,
   GossipSerializer,
   GossipType,
   GossipTopic,
-  GossipTopicMap,
+  GossipTopicTypeMap,
 } from "./interface";
-import {DEFAULT_ENCODING} from "./constants";
+
+const topicPrefix = "/eth2";
 
 /**
- * Create a gossip topic string
+ * Stringify a GossipTopic into a spec-ed formated topic string
  */
-export function getGossipTopicString(forkDigestContext: IForkDigestContext, topic: GossipTopic): string {
+export function stringifyGossipTopic(forkDigestContext: IForkDigestContext, topic: GossipTopic): string {
   const forkDigest = forkDigestContext.forkName2ForkDigest(topic.fork);
-  const forkDigestHex = toHexString(forkDigest).toLowerCase().substring(2);
-  let topicType: string = topic.type;
-  if (topic.type === GossipType.beacon_attestation) {
-    topicType += "_" + topic.subnet;
-  }
-  if (topic.type === GossipType.sync_committee) {
-    topicType += "_" + topic.subnet;
-  }
-  return `/eth2/${forkDigestHex}/${topicType}/${topic.encoding ?? DEFAULT_ENCODING}`;
+  const forkDigestHexNoPrefix = toHexStringNoPrefix(forkDigest);
+  const topicType = stringifyGossipTopicType(topic);
+  return [topicPrefix, forkDigestHexNoPrefix, topicType, topic.encoding ?? DEFAULT_ENCODING].join("/");
 }
 
-// eslint-disable-next-line @typescript-eslint/naming-convention
-export const GossipTopicRegExp = new RegExp("^(/eth2/)([a-f0-9]{8})/(\\w+)/(\\w+)");
+/**
+ * Parse a `GossipTopic` object from its stringified form.
+ * A gossip topic has the format
+ * ```ts
+ * /eth2/$FORK_DIGEST/$GOSSIP_TYPE/$ENCODING
+ * ```
+ */
+export function parseGossipTopic(forkDigestContext: IForkDigestContext, topicStr: string): GossipTopic {
+  try {
+    if (!topicStr.startsWith(topicPrefix)) {
+      throw Error(`Must be prefixed with ${topicPrefix}`);
+    }
 
-// eslint-disable-next-line @typescript-eslint/naming-convention
-export const AttestationSubnetRegExp = new RegExp("^/eth2/[a-f0-9]{8}/beacon_attestation_([0-9]+)/\\w+$");
+    const topicStrNoPrefix = topicStr.slice(topicPrefix.length + 1); // +1 for the second '/' in '/eth2/...'
+    const [forkDigestHexNoPrefix, gossipTypeStr, encodingStr] = topicStrNoPrefix.split("/");
```

Comment:
```suggestion
    const [,,forkDigestHexNoPrefix, gossipTypeStr, encodingStr] = topicStrNoPrefix.split("/");
```
I think that now you can remove slice line above. Maybe it will be a little bit easier to read code.

Reply:
But it's not the same logic, you are doing extra work since you have to run .split() on extra characters that you already now are not useful.

Reply:
do you think that split vs slice would have notable performance impact?

Reply:
> do you think that split vs slice would have notable performance impact?

Maybe, I've not quantified it. Tho this is a hot path since we want to process a lot of gossip objects per second. To understand your proposal do you suggest
```ts
const [,,forkDigestHexNoPrefix, gossipTypeStr, encodingStr] = topicStrNoPrefix.split("/");
```
or
```ts
const [,,forkDigestHexNoPrefix, gossipTypeStr, encodingStr] = topicStr.split("/");
```
?

Assuming the latter I don't see the advantage of the `[,,` syntax towards readability, such that it justifies doing extra unnecessary work. I can add more comments to make the intention and rationale more clear.

Reply:
I've benchmark it and
- splice is very fast, 250000 opts / ms
- split is costlier at 2000 opts / ms

Using a regex is slightly faster to extract topic parts (~10%) so I've updated that.